src/internal_events/dedupe.rs

```diff
@@ -2,13 +2,18 @@ use metrics::counter;
 use vector_core::internal_event::InternalEvent;
 
 #[derive(Debug)]
-pub struct DedupeEventDiscarded {
-    pub event: crate::event::Event,
+pub struct DedupeEventsDropped<const N: usize> {
+    pub events: [crate::event::Event; N],
 }
 
-impl InternalEvent for DedupeEventDiscarded {
+impl<const N: usize> InternalEvent for DedupeEventsDropped<N> {
     fn emit(self) {
-        trace!(message = "Encountered duplicate event; discarding.", event = ?self.event);
-        counter!("events_discarded_total", 1);
+        debug!(
+            message = "Events dropped.",
+            count = self.events.len(),
+            intentional = true,
+            reason = "Events have been found in cache for deduplication."
+        );
+        counter!("events_discarded_total", self.events.len() as u64);
```

Comment:
👍 I don't think this quite matches the spec though, the `counter!` should have the `intentional` tag as well.